cpp/src/parquet/arrow/arrow_schema_test.cc

```diff
@@ -644,6 +649,78 @@ TEST_F(TestConvertParquetSchema, ParquetRepeatedNestedSchema) {
   ASSERT_NO_FATAL_FAILURE(CheckFlatSchema(arrow_schema));
 }
 
+TEST_F(TestConvertParquetSchema, ColumnSubselection) {
+  std::vector<NodePtr> parquet_fields;
+  std::vector<std::shared_ptr<Field>> arrow_fields;
+  {
+    //   optional int32 leaf1;
+    //   repeated group outerGroup {
+    //     optional int32 leaf2;
+    //     repeated group innerGroup {
+    //       optional int32 leaf3;
+    //     }
+    //   }
+    parquet_fields.push_back(
+        PrimitiveNode::Make("leaf1", Repetition::OPTIONAL, ParquetType::INT32));
+    parquet_fields.push_back(GroupNode::Make(
+        "outerGroup", Repetition::REPEATED,
+        {PrimitiveNode::Make("leaf2", Repetition::OPTIONAL, ParquetType::INT32),
+         GroupNode::Make(
+             "innerGroup", Repetition::REPEATED,
+             {PrimitiveNode::Make("leaf3", Repetition::OPTIONAL, ParquetType::INT32)})}));
+
+    auto inner_group_fields = {::arrow::field("leaf3", INT32, true)};
```

Comment:
please spell out types in some form where you are using bracket initializations.

Reply:
P.S. we really need a JSON utility to construct a schema from json., it would make this much easier to read.

Reply:
Re JSON util: I'm not convinced that would be significantly more concise than our (de-facto) DSL with `schema`, `field`, and friends. Here's what I'm thinking of:

```json
{
  "leaf1": "int32",
  "outerGroup": {
    "nested": "struct",
    "fields": [{
      "leaf2": "int32",
      "innerGroup": {
        "nested": "list not null",
        "fields": [{
          "inner-erGroup": {
            "nested": "struct not null",
            "fields": {
              "leaf3": "int32"
            }
          }
        }]
      }
    }]
  }
}
```

vs

```c++
schema({
  field("leaf1", int32()),
  field("outerGroup", struct_({
    field("leaf2", int32()),
    field("innerGroup", list(
      field("inner-erGroup", struct_({
        field("leaf3", int32())
      }), false)
    ), false),
  }))
})
```

Reply:
https://issues.apache.org/jira/browse/ARROW-9422